dlib/type_safe_union/type_safe_union_kernel.h

```diff
@@ -570,6 +514,116 @@ namespace dlib
         type_safe_union<Types...>& b
     ) { a.swap(b); }
 
+    namespace detail
+    {
+        template<
+            typename Variant,
+            typename F,
+            std::size_t... I
+        >
+        void for_each_type_impl(
+            Variant&& tsu,
+            F&& f,
+            dlib::index_sequence<I...>
+        )
+        {
+            using Tsu = typename std::decay<Variant>::type;
+            return (void)std::initializer_list<int>{
```

Comment:
```suggestion
            int unused[] = {
```
I think this is a simpler way to write this kind of thing, since it's less text and avoids that `return` there that looks like it should be returning something but is actually in a `void` function.  Also avoids the `(void)` cast which is squirrely in general.

Reply:
yes this does look a bit nicer but i'm worried that:
- people will get excessive warnings about unused variables
- compilers who are too smart for their own good will optimize out the entire thing because `unused` is unused.

Reply:
Huh, yeah I suppose those are possabilities.  So keep what you have :+1: